src/resolver.rs

```diff
@@ -141,4 +167,40 @@ mod tests {
         };
         resolve::<EVMDialect>(&mut ast);
     }
+
+    use crate::yul_parser::{parse_block, parse_expression};
+
+    #[test]
```

Comment:
Is it common to have tests inside an impl?

Reply:
There is `mod test` in line 150, this is not inside the impl.